### PR TITLE
vsphere: machine-config add upgrade

### PR DESCRIPTION
**Pull request by jcpowermac**

### CODE REVIEW

**Comment by jcpowermac** [comment]
MCO changes seem to be where a majority of our vSphere upgrade failures happen. Let's get that upgrade job available to run ;-)

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.6-presubmits.yaml [hunk 1]
```diff
@@ -899,6 +899,72 @@ presubmits:
...
+    cluster: vsphere
+    context: ci/prow/e2e-vsphere-upgrade
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/prowgen-controlled: "true"
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-machine-config-operator-release-4.6-e2e-vsphere-upgrade
+    rerun_command: /test e2e-vsphere-upgrade
```
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.6-presubmits.yaml:915]: add optional: true

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.7-presubmits.yaml [hunk 2]
```diff
@@ -1170,6 +1170,72 @@ presubmits:
...
+    cluster: vsphere
+    context: ci/prow/e2e-vsphere-upgrade
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/prowgen-controlled: "true"
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-machine-config-operator-release-4.7-e2e-vsphere-upgrade
+    rerun_command: /test e2e-vsphere-upgrade
```
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.7-presubmits.yaml:1186]: optional: true

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.8-presubmits.yaml [hunk 3]
```diff
@@ -1170,6 +1170,72 @@ presubmits:
...
+    cluster: vsphere
+    context: ci/prow/e2e-vsphere-upgrade
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/prowgen-controlled: "true"
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-machine-config-operator-release-4.8-e2e-vsphere-upgrade
+    rerun_command: /test e2e-vsphere-upgrade
```
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.8-presubmits.yaml:1186]: optional: true

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.9-presubmits.yaml [hunk 4]
```diff
@@ -1170,6 +1170,72 @@ presubmits:
...
+    cluster: vsphere
+    context: ci/prow/e2e-vsphere-upgrade
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/prowgen-controlled: "true"
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-machine-config-operator-release-4.9-e2e-vsphere-upgrade
+    rerun_command: /test e2e-vsphere-upgrade
```
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.9-presubmits.yaml:1186]: optional: true

**Review by kikisdeliveryservice — CHANGES_REQUESTED** [review]
add optional: true,
make your changes (with optional) in master.yaml

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.6-presubmits.yaml (hunk 1 above)
Comment (kikisdeliveryservice) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.6-presubmits.yaml:915]: also wondering why there is no 4.6 dry run below in prow ? :thinking:

Comment (jcpowermac) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.6-presubmits.yaml:916]: @kikisdeliveryservice its there it just succeeded :D

Comment (kikisdeliveryservice) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-release-4.6-presubmits.yaml:916]: ahhhh hahaha my bad :laughing:

**Review by kikisdeliveryservice — APPROVED** [review]

**Pull request merged**